.husky/pre-commit

```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
+
+lint-staged
```

Comment:
when I run the command `git commit -m "some commit"` it shows the below error 
```
.husky/pre-commit: 3: lint-staged: not found
``` 
It works after adding `yarn lint-staged` 

Then the command run but shows an error and the changes are not committed 
Error : ✖   type may not be empty [type-empty]
```
@elhaw ➜ /workspaces/m3ntorship-ui (initial-project-setup) $ git commit -m "added  css styles"
yarn run v1.22.15
$ /workspaces/m3ntorship-ui/node_modules/.bin/lint-staged
[STARTED] Preparing lint-staged...
[SUCCESS] Preparing lint-staged...
[STARTED] Running tasks for staged files...
[STARTED] package.json — 5 files
[STARTED] src/**/*.{js,jsx,ts,tsx} — 1 file
[STARTED] src/**/*.{json,css,scss,md} — 0 files
[SKIPPED] src/**/*.{json,css,scss,md} — no files
[STARTED] tsdx lint --max-warnings=0
[SUCCESS] tsdx lint --max-warnings=0
[STARTED] prettier -w
[SUCCESS] prettier -w
[SUCCESS] src/**/*.{js,jsx,ts,tsx} — 1 file
[SUCCESS] package.json — 5 files
[SUCCESS] Running tasks for staged files...
[STARTED] Applying modifications from tasks...
[SUCCESS] Applying modifications from tasks...
[STARTED] Cleaning up temporary files...
[SUCCESS] Cleaning up temporary files...
Done in 5.91s.
⧗   input: added  css styles
✖   type may not be empty [type-empty]

✖   found 1 problems, 0 warnings
ⓘ   Get help: https://github.com/conventional-changelog/commitlint/#what-is-commitlint

husky - commit-msg hook exited with code 1 (error)
```

if there some conventions we need to introduce it to the whole team so that every one is aware if it.

Reply:
Yes, the commit message structure is explained by the written rules in `commitlint.config.js` you can take a look at the predefined words which should be included in the commit message، 

And for the `lint-staged` command,  we can install lint-staged globally on our machines or by just adding `yarn lint-staged` in the pre-commit hook file

Reply:
rule of thumb is to depend  on local packages only 
refer to this article [](https://www.smashingmagazine.com/2016/01/issue-with-global-node-npm-packages/)